src/main/java/org/apache/commons/text/TextStringBuilder.java

```diff
@@ -1947,7 +1954,7 @@ public String getNullText() {
      */
     @Override
     public int hashCode() {
-        return Arrays.hashCode(buffer);
+        return Arrays.hashCode(ArrayUtils.subarray(buffer, 0, size));
```

Comment:
I think we can reuse our own toString() here.

Reply:
indeed, done in new commit